test/test_backend.py

```diff
@@ -225,6 +233,154 @@ def test_backend_cmdline(self):
         t = _TesterCmdline(None, ['-v'])
         self.assertTrue(t.args.verbose)
 
+    def test_resolve_aliases(self):
+        first_alias = Alias(None, None, None)
+        first_alias.data_type = String()
+        returned_alias = resolve_aliases(first_alias)
+
+        # Test that single-level alias chain resolves
+        self.assertEqual(returned_alias, first_alias)
+        self.assertIsInstance(returned_alias.data_type, String)
+
+        first_alias = Alias(None, None, None)
+        second_alias = Alias(None, None, None)
+        first_alias.data_type = second_alias
+        second_alias.data_type = String()
+
+        # Test that a two-level alias chain resolves
+        returned_alias = resolve_aliases(first_alias)
+        self.assertEqual(returned_alias, first_alias)
+        self.assertIsInstance(first_alias.data_type, String)
+        self.assertIsInstance(second_alias.data_type, String)
+
+    def test_preserve_aliases_from_api(self):
+        spec = """\
+namespace preserve_alias
+
+route eval(TestStruct, TestResult, TestError)
+
+alias NamespaceId = String(pattern="[-_0-9a-zA-Z:]+")
+alias SharedFolderId = NamespaceId
+alias PathRootId = SharedFolderId
+
+struct TestStruct
+    field1 PathRootId
+
+alias StructAlias = TestStruct
+
+struct TestResult
+    field1 PathRootId
+
+union TestError
+    test PathRootId
+"""
+
+        api = specs_to_ir(
```

Comment:
Can we construct IR directly instead of from the Stone code, like how it's done in `test_api_namespace`? This removes the dependency on frontend code so that the tests won't fail due to bugs there.